src/modules/workspaces/workspaces.controller.spec.ts

```diff
@@ -240,5 +240,73 @@ describe('Workspace Controller', () => {
       );
       expect(getMembers).toEqual(mockResolvedValues);
     });
+
+    describe('GET /invitations/:inviteId/validate', () => {
+      const mockResponse = {
+        status: jest.fn().mockReturnThis(),
+        send: jest.fn().mockReturnThis(),
+      };
+      it('When invitation is validated successfully, then it returns.', async () => {
+        const invite = newWorkspaceInvite();
+        workspacesUsecases.validateWorkspaceInvite.mockResolvedValueOnce(
+          invite.id,
+        );
+
+        const validateInvite =
+          await workspacesController.validateWorkspaceInvitation(
+            invite.id,
+            mockResponse as any,
+          );
+
+        expect(workspacesUsecases.validateWorkspaceInvite).toHaveBeenCalledWith(
+          invite.id,
+        );
+        expect(validateInvite).toEqual(invite.id);
+      });
+
+      it('When invitation is validated with invalid inviteId, then it should throw BadRequestException', async () => {
+        const errorMessage = 'Invalid invitation ID';
+        workspacesUsecases.validateWorkspaceInvite.mockRejectedValueOnce(
+          new BadRequestException(errorMessage),
+        );
+
+        try {
+          await workspacesController.validateWorkspaceInvitation(
+            'invalid-id',
+            mockResponse as any,
+          );
+          // If the above line does not throw, the test should fail.
+          fail(
```

Comment:
Fail was removed from jest since version 27. You can remove it as it will just throw because "fail is undefined". I do not know why @types/jest still marks this as valid.